### DISCUSSION TITLE
Like to spawn a command at start.

### DISCUSSION THREAD

**Discussion by moilong** [YaLTeR/niri#339] [Q&A] [answered]
I like to start swww to set wallpaper in niri. My config looks like this:
spawn-at-startup "swww" "init"
spawn-at-startup "swww" "img" "$(< "${HOME}/.cache/wal/wal")"
First one to start swww-daemon. Works ok.
Second one to set wallpaper not executed. I tried running as bash script. 
spawn-at-startup "swww.sh". 
Again no worky. As a workaround I put that second command in .bash_profile. Help appreciated. Thankyou.

**Comment by YaLTeR** [comment, marked as answer]
They are started in parallel, so likely swww img starts before the daemon has a chance to start up. I suggest putting these two in a single shell script or command so that they run sequentially.

**Reply by moilong** [reply]
Yes I tried that like
swww init && swww img "$(< "${HOME}/.cache/wal/wal")"
and the script also not executed.

**Reply by moilong** [reply]
Script like this works:
`swww init && swww img ~/Pictures/niribg/leaves.jpg`
